### PR TITLE
Carlene/mixpanel

### PR DESCRIPTION
**Pull request by carlenee**
#### What's New

N/A

#### What's Updated

Mix panel added on these features : 
- Open Calculator
- Add Course on Calculator
- View Course on Calculator
- Add Course Component on Calculator
- Delete Course from Calculator
- Open Profile
- Saved Matkuls from Profile

#### New Package Added

```yaml
"dependencies": 
  - // N / A

"dev_dependencies": 
  - // N / A

```

### Recommendations on how to test the changes

N / A

### Checklist for developer

- [v ] The code builds clean without any errors or warnings
- [v ] The code does not contain commented out code
- [ ] The code does not log anything to console
- [ ] I have added unit test(s) to cover new code and successfully executed ran it
- [ ] I have thoroughly tested the new code and any adjacent features it may affect

### Link to related issue(s)

N / A

#### Before

N / A

#### After

N / A

### CODE REVIEW

File: lib/features/kalkulator/presentation/pages/kalkulator_page.dart [hunk 1]
```diff
@@ -116,19 +119,24 @@ Kamu Belum memiliki kalkulator nilai tersimpan. Silakan tambahkan terlebih dahul
...
+                            totalScore: calculator.totalScore!,
+                            totalPercentage: calculator.totalPercentage!,
                           );
+                          MixpanelService.track('calculator_view_course',
+                              params: {
+                                'course_id': 'CSCM12349990',
+                                'final_letter_grade': 'A',
+                                'final_grade': '88.75',
+                              },);
                         },
```
Comment (yudha-haris) [inline lib/features/kalkulator/presentation/pages/kalkulator_page.dart]: this params isn't suppose to be hardcoded
please refer to the variable, for example

'course_id': calculator.id,
'final_grade': calculator.totalScore,
etc..

File: lib/features/kalkulator/presentation/pages/komponen_kalkulator_page.dart [hunk 2]
```diff
@@ -184,6 +184,14 @@ class _CalculatorComponentPageState
                         QueryCalculator(id: widget.calculatorId),
                       ),
                     );
+                    MixpanelService.track(
+                      'calculator_delete_course_component',
+                      params: {
+                        'course_id': 'CSCM12349990',
+                        'final_letter_grade': 'A',
+                        'final_grade': '88.75',
+                      },
```
Comment (yudha-haris) [inline lib/features/kalkulator/presentation/pages/komponen_kalkulator_page.dart]: this params isn't suppose to be hardcoded
please refer to the variable, for example

'course_id': calculator.id,
'final_grade': calculator.totalScore,
etc..

**Review by yudha-haris — CHANGES_REQUESTED** [review]

File: lib/features/kalkulator/presentation/pages/komponen_kalkulator_page.dart (hunk 2 above)
Comment (veloraine) [reply lib/features/kalkulator/presentation/pages/komponen_kalkulator_page.dart]: regarding the params for the mixpanel event for calculator deletion, use params that available on the calculator component page, on this case:

1. 'course_name': widget.courseName
2. 'final_grade': widget.totalScore
3. 'final_letter_grade': _getFinalScoreAndGrade(widget.totalScore)

**Review by yudha-haris — APPROVED** [review]

File: lib/features/kalkulator/presentation/pages/kalkulator_page.dart [hunk 3]
```diff
@@ -159,4 +170,27 @@ Kamu Belum memiliki kalkulator nilai tersimpan. Silakan tambahkan terlebih dahul
...
+    } else if (score >= 65) {
+      grade = 'B-';
+    } else if (score >= 60) {
+      grade = 'C+';
+    } else if (score >= 55) {
+      grade = 'C';
+    } else if (score >= 40) {
+      grade = 'D';
+    }
+
```
Comment (yudha-haris) [inline lib/features/kalkulator/presentation/pages/kalkulator_page.dart]: use switch case instead

File: lib/features/kalkulator/presentation/pages/komponen_kalkulator_page.dart [hunk 4]
```diff
@@ -239,6 +249,29 @@ class _CalculatorComponentPageState
...
+      grade = 'B';
+    } else if (score >= 65) {
+      grade = 'B-';
+    } else if (score >= 60) {
+      grade = 'C+';
+    } else if (score >= 55) {
+      grade = 'C';
+    } else if (score >= 40) {
+      grade = 'D';
+    }
```
Comment (yudha-haris) [inline lib/features/kalkulator/presentation/pages/komponen_kalkulator_page.dart]: use switch case instead

if used in multiple pages, consider write in utils

DRY principle = Dont Repeat Yourself

File: lib/main_page.dart [hunk 5]
```diff
@@ -70,7 +71,14 @@ class _MainPageState extends BaseStateful<MainPage> {
...
     return RistekBotNavBar(
       initialActiveIndex: _selectedIndex,
-      onTap: (int index) => setState(() => _selectedIndex = index),
+      onTap: (int index) {
+        setState(() => _selectedIndex = index);
+        if (index == 3) {
+          MixpanelService.track('open_calculator');
+        } else if (index == 4) {
+          MixpanelService.track('open_profile');
+        }
```
Comment (yudha-haris) [inline lib/main_page.dart]: switch (index) {
          case 1:
            MixpanelService.track('open_courses');
          case 2:
            MixpanelService.track('open_klasemen');
          case 3:
            MixpanelService.track('open_calculator');
          case 4:
            MixpanelService.track('open_profile');
        }

Comment (yudha-haris) [reply lib/main_page.dart]: use switch case to make code more readable

**Pull request merged**